pyqmc/linemin.py

```diff
@@ -47,24 +47,33 @@ def polyfit_relative(xfit, yfit, degree):
     return p, relative_error
 
 
-def stable_fit3(xfit, yfit, tolerance=1e-2):
-    """Try to fit to a quadratic. If the fit is not good,
-    then just take the lowest value of yfit
+def stable_fit(xfit, yfit, tolerance=1e-2):
+    """Fit a line and quadratic to xfit and yfit. 
+    1. If the linear fit is as good as the quadriatic, choose the lower endpoint.
+    2. If the curvature is positive, estimate the minimum x value.
+    3. If the lowest yfit is less than the new guess, use that xfit instead.
+
+    :parameter list xfit: scalar step sizes along line
+    :parameter list yfit: estimated energies at xfit points
+    :parameter float tolerance: how good the quadratic fit needs to be
+    :returns: estimated x-value of minimum
+    :rtype: float
     """
     steprange = np.max(xfit)
     minstep = np.min(xfit)
     a = np.argmin(yfit)
     pq, relative_errq = polyfit_relative(xfit, yfit, 2)
     pl, relative_errl = polyfit_relative(xfit, yfit, 1)
+    if 0. not in xfit:
```

Comment:
I think we should not do this. It checks equality on a float, and steps out of the lane anyway, in my opinion.